Example/Tests/BONUtilitiesTestCase.m

```diff
@@ -0,0 +1,254 @@
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+- (void)testTextViewTextBeforeChainable
+{
+    UITextView *textView = UITextView.new;
+    textView.textAndApplyChainable = @"Hello, world!";
+    textView.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertEqualObjects(textView.attributedText.string, @"Hello, world!");
+
+    NSParagraphStyle *defaultParagraphStyle = [[NSParagraphStyle alloc] init];
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+- (void)testTextViewAttributedTextAfterChainable
+{
+    UITextView *textView = UITextView.new;
+    textView.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textView.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+
+    XCTAssertEqualObjects(textView.attributedText.string, @"Hello, world!");
+
+    NSDictionary *controlAttributes = [self defaultAttributesForClass:[UITextView class] withString:@"Hello, world!"];
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+- (void)testTextViewAttributedTextBeforeChainable
+{
+    UITextView *textView = UITextView.new;
+    textView.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+    textView.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertEqualObjects(textView.attributedText.string, @"Hello, world!");
+
+    NSParagraphStyle *defaultParagraphStyle = [[NSParagraphStyle alloc] init];
+
+    NSDictionary *controlAttributes = @{
+        BONValueFromRange(0, 13) : @{
+            NSParagraphStyleAttributeName : defaultParagraphStyle,
+            NSFontAttributeName : [UIFont systemFontOfSize:16],
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+#pragma mark - UITextField
+
+- (void)testTextFieldTextAfterChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textField.textAndApplyChainable = @"Hello, world!";
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
```

Comment:
#nth in the interest of deterministic tests that fail with nice log messages, `XCTAssertNotNil(controlAttributes[BONValueFromRange(0, 13)])` before this line